nrrd/reader.py

```diff
@@ -339,48 +339,87 @@ def read_data(header, fh=None, filename=None):
     # Get the total number of data points by multiplying the size of each dimension together
     total_data_points = header['sizes'].prod()
 
-    # If encoding is raw and byte skip is -1, then seek backwards to the data
-    # Otherwise skip the number of lines requested
-    if header['encoding'] == 'raw' and byte_skip == -1:
-        fh.seek(-dtype.itemsize * total_data_points, 2)
-    else:
-        for _ in range(line_skip):
-            fh.readline()
-
-    # If a compression encoding is used, then byte skip AFTER decompressing
-    if header['encoding'] == 'raw':
-        # Skip the requested number of bytes and then parse the data using NumPy
-        fh.seek(byte_skip, os.SEEK_CUR)
-        data = np.fromfile(fh, dtype)
-    elif header['encoding'] in ['ASCII', 'ascii', 'text', 'txt']:
-        # Skip the requested number of bytes and then parse the data using NumPy
-        fh.seek(byte_skip, os.SEEK_CUR)
-        data = np.fromfile(fh, dtype, sep=' ')
-    else:
-        # Handle compressed data now
-        # Construct the decompression object based on encoding
-        if header['encoding'] in ['gzip', 'gz']:
-            decompobj = zlib.decompressobj(zlib.MAX_WBITS | 16)
-        elif header['encoding'] in ['bzip2', 'bz2']:
-            decompobj = bz2.BZ2Decompressor()
+    # Skip the number of lines requested
+    for _ in range(line_skip):
+        fh.readline()
+
+    if byte_skip>=0:
+    
```

Comment:
Although this code functionally works correctly, I think it can be made much cleaner. You have a large if/else statement with a lot of repeated code. I would stick to the principle of [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

In other words, a cleaner way is to just change what needs to be changed.